### ISSUE TITLE
Onion domains

### ISSUE THREAD

**Issue by rugk** [PrivateBin/Directory#2]
There were quite some bins that (alos) offer onion domains.

How do we deal with these?
Either integrate Tor support (hard) or do not check them(?).

**Comment by elrido** [comment]
They are currently not supported, because:

1. .onion domains don't resolve on the root DNS servers, so it fails when looking up the IP of the given hostname (it's therefore considered an invalid hostname).
2. checking would require access to a tor client in proxy mode. Using such a setup would let us use the tor DNS, which also would solve 1. and let us access it. But this is a tricky setup and I have contractual obligations not to host tor exits or route tor traffic by my server housing provider, so for now it's out of scope.

**Comment by rugk** [comment]
Yeah, I'd already think it is out-of scope. So maybe there should be a way (for admins as you, at least) to manually add instances (and their status)?

**Comment by elrido** [comment]
I rather have this fully automated and low maintenance. I had been considering to allow admins to add custom descriptions via a static text file or configuration option in some new API endpoint on their instance - maybe a feature for 1.4 in the scope of the federation ticket.

**Comment by cbluth** [comment]
how can i find some onion domains? i would like to test my cli client on them.

**Comment by elrido** [comment]
There are currently two onion domains listed in the [wiki directory](https://github.com/PrivateBin/PrivateBin/wiki/PrivateBin-Instances-Directory).

**Comment by NoPlagiarism** [comment]
> There are currently two onion domains

Now there are 3 domain on wiki

Anyway, [I found 5 onion domains and 1 .i2p using header parsing](https://github.com/NoPlagiarism/instances-list/tree/master/instances/tools/privatebin#onion)

Listing them here (10 Dec 2023 17:41:47 UTC):
- [5b6kg6vyo4uk2w7y6eibyuhvpoxnkyaxkjqo72pomcbgbqfmenas3eqd.onion](http://5b6kg6vyo4uk2w7y6eibyuhvpoxnkyaxkjqo72pomcbgbqfmenas3eqd.onion)
- [n63ite5off46lfh7qei4uhkvttrgvpve7ag3kwftlqkxo4o5mu7l4cqd.onion](http://n63ite5off46lfh7qei4uhkvttrgvpve7ag3kwftlqkxo4o5mu7l4cqd.onion)
- [paste.ona32cmhotftdnfrsyxoopqjjbev6jhy3gdujlozqj2altj5mhrn4lyd.onion](http://paste.ona32cmhotftdnfrsyxoopqjjbev6jhy3gdujlozqj2altj5mhrn4lyd.onion)
- [qibpme6jnnxjjyqvwfnioes4vkplyrixjd7vopuzl3xsxqnzjhlhvjqd.onion](http://qibpme6jnnxjjyqvwfnioes4vkplyrixjd7vopuzl3xsxqnzjhlhvjqd.onion)
- [z2hx4persafjkxw3u23dly24tzzynvkzrp5r3zbh6jrjvd7c7qjsgvyd.onion](http://z2hx4persafjkxw3u23dly24tzzynvkzrp5r3zbh6jrjvd7c7qjsgvyd.onion)
